src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -12,16 +14,99 @@ export class AbstractQueryInterfaceTypeScript {
   }
 
   /**
-   * Creates a new database schema.
+   * Create a new database schema.
    *
    * **Note:** this is a schema in the [postgres sense of the word](http://www.postgresql.org/docs/9.1/static/ddl-schemas.html),
-   * not a database table. In mysql and sqlite, this command will do nothing.
+   * not a database table. In mysql and mariadb, this command will instead create a database.
    *
-   * @param schema
-   * @param options
+   * @see
+   * {@link Model.schema}
+   *
+   * @param schema Name of the schema
+   * @param [options={}]
+   * @param [options.collate=null]
+   * @param [options.charset=null]
+    *
+   * @returns
    */
   async createSchema(schema: string, options?: CreateSchemaOptions): Promise<void> {
     const sql = this.queryGenerator.createSchemaQuery(schema, options);
     await this.sequelize.queryRaw(sql, options);
   }
+
+  /**
+   * Drop a single schema
+   *
+   * **Note:** this is a schema in the [postgres sense of the word](http://www.postgresql.org/docs/9.1/static/ddl-schemas.html),
+   * not a database table. In mysql and sqlite, this drop a table matching the schema name
+   *
+   * @param schema Name of the schema
+   * @param [options={}] query options
+   * @param [options.logging] A function that logs sql queries, or false for no logging
+   *
+   * @returns
+   */
+  async dropSchema(schema: string, options?: QueryRawOptions): Promise<void> {
+    const dropSchemaQuery: string | QueryGeneratorDropSchemaQueryObject = this.queryGenerator.dropSchemaQuery(schema);
+
+    let sql: string;
+    let queryRawOptions = options;
+    if (typeof dropSchemaQuery === 'string') {
+      sql = dropSchemaQuery;
```

Comment:
```suggestion
    let queryRawOptions: QueryRawOptions;
    if (typeof dropSchemaQuery === 'string') {
      sql = dropSchemaQuery;
      queryRawOptions = options;
```

Reply:
Conversation in https://github.com/sequelize/sequelize/pull/15498#discussion_r1070651102